src/audio/base_fw.c

```diff
@@ -44,6 +44,25 @@ DECLARE_TR_CTX(basefw_comp_tr, SOF_UUID(basefw_uuid), LOG_LEVEL_INFO);
 static struct ipc4_system_time_info global_system_time_info;
 static uint64_t global_cycle_delta;
 
+__cold static uint32_t get_host_buffer_size(void)
+{
+	struct sof_dma *dma_host;
+	uint32_t periods;
```

Comment:
we can omit calling `assert_can_be_cold()` if this is a `static` function and all the callers have called it - as is the case here. But let's at least make a comment about it to make sure this isn't broken in the future.

Reply:
a comment would not stop misuse of this function, which is a helper to be used by `basefw_config()`
I will add the assert_can_be_cold() instead of a comment.